autopilot/hostinfo.go

```diff
@@ -10,7 +10,10 @@ import (
 )
 
 func (c *contractor) HostInfo(ctx context.Context, hostKey types.PublicKey) (api.HostHandlerGET, error) {
-	cfg := c.ap.Config()
+	cfg, err := c.ap.Config(ctx)
```

Comment:
Since we already keep the current period in memory why not do the whole config? To avoid fetching it more than once per iteration at most.

Reply:
Well it's because we keep _loop_ state, which isn't thread safe because we only have one thread doing the autopilot loop. Everything we spin up a goroutine for is launched with the state at the time. This is why I fetch the `Config` from the bus here.

Yesterday though I was considering extending the `state` with the current period, so then it would have everything but the downside would be that the autopilot would manage the period, not the contractor. Which honestly isn't that much of an issue imo. I'll try that and see where I end up, might be cleaner over all and we can use a cached config here.